DuckDuckGo/Onboarding/ContextualOnboarding/OnboardingFireButtonDialogViewModel.swift

```diff
@@ -0,0 +1,33 @@
+//
+//  OnboardingFireButtonDialogViewModel.swift
+//
+//  Copyright © 2024 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+
+public struct OnboardingFireButtonDialogViewModel {
+    let onDismiss: () -> Void
+
+    func skip() {
+        onDismiss()
+    }
+
+    func tryFireButton() {
+        DispatchQueue.main.async {
```

Comment:
Shouldn’t we already be on the MainThread if we call `tryFireButton` from the UI ?  Alternatively we could mark the VM @MainActor?